Source/WebCore/html/PermissionsPolicy.cpp

```diff
@@ -123,315 +232,148 @@ bool isPermissionsPolicyAllowedByDocumentAndAllOwners(PermissionsPolicy::Type ty
     return true;
 }
 
-static bool isAllowedByPermissionsPolicy(const PermissionsPolicy::AllowRule& rule, const SecurityOriginData& origin)
+void PermissionsPolicy::Allowlist::addOrigin(const SecurityOriginData& origin)
 {
-    switch (rule.type) {
-    case PermissionsPolicy::AllowRule::Type::None:
-        return false;
-    case PermissionsPolicy::AllowRule::Type::All:
+    if (HashSet<SecurityOriginData>* origins = std::get_if<HashSet<SecurityOriginData>>(&m_origins))
+        origins->add(origin);
```

Comment:
This is a bit strange, auto* might be slightly better,.
The if might not be needed in practice?
I would tend to build a HashSet<SecurityOriginData>, then once filled, create an AllowList from it.

Reply:
I don't quite understand this comment; are you suggesting `Allowlist` should own a `HashSet` instead of `std::variant`?

Reply:
I would remove addOrigin altogether and add an `explicit PermissionsPolicy::Allowlist(HashSet&&)`

Reply:
ok will update